### PR TITLE
Fix unloaded files merge

### PR DESCRIPTION
**Pull request by hugopeixoto**
A test case and an attempt at fixing #433

### CODE REVIEW

**Comment by xaviershay** [comment]
I'm not familiar enough to review this properly - the model seems weird but this change seems reasonable without a more extensive refactoring (if that's even appropriate).

Will wait for either someone else to chime in or a confirmation that this has fixed #433 before merging.

Thank you!

File: spec/merge_helpers_spec.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@
       end
 
       it "has proper results for sample.rb" do
-        expect(subject[source_fixture("sample.rb")]).to eq([1, 1, 2, 2, nil, nil, 2, 2, nil, nil])
+        expect(subject[source_fixture("sample.rb")]).to eq([nil, nil, 2, 2, nil, nil, 2, 2, nil, nil])
```
Comment (sinsoku) [inline spec/merge_helpers_spec.rb]: Why did you change the existing behavior?
This will occur a problem when we run the tests in parallel. Because covered line become a disabled line.

**Comment by bfontaine** [comment]
Just FYI there’s a different fix for the same issue:
```patch
diff --git i/lib/simplecov/merge_helpers.rb w/lib/simplecov/merge_helpers.rb
index 339210d1..ca6b8a9f 100644
--- i/lib/simplecov/merge_helpers.rb
+++ w/lib/simplecov/merge_helpers.rb
@@ -2,6 +2,12 @@ module SimpleCov
   module ArrayMergeHelper
     # Merges an array of coverage results with self
     def merge_resultset(array)
+      # Don't merge with all-zero array to preserve nil values from actual runs.
+      if size == array.size
+        return array.dup if all? { |el| el == 0 }
+        return dup if array.all? { |el| el == 0 }
+      end
+
       new_array = dup
       array.each_with_index do |element, i|
         if element.nil? && new_array[i].nil?
```

See https://github.com/Homebrew/homebrew/pull/48250#issuecomment-173171990.

**Comment by xaviershay** [comment]
Closing in favour of #441
